File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        // 01.11.24 POTD
        string ans;
        int count=1;

        for(auto c: s){
            if(!ans.empty() && c == ans.back())
                count++;
            else
                count = 1;

            if(count<3)
                ans += c;
        }

        return ans;
    }
};

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        # 21.07.25 POTD
        ans = ""
        cnt = 0
        prev = s[0]

        for c in s:
            if prev == c:
                cnt += 1
            else:
                cnt = 1
                prev = c
            
            if cnt < 3:
                ans += c
        
        return ans
